aws/gpu_tree_shap/gpu_tree_shap.ipynb

```diff
@@ -0,0 +1,479 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# GPU-Accelerated SHAP values with XGBoost 1.3 on AWS\n",
+    "\n",
+    "With the release of XGBoost 1.3 comes an exciting new feature for model interpretability — GPU accelerated SHAP values. SHAP values are a technique for local explainability of model predictions. That is, they give you the ability to examine the impact of various features on model output in a principled way. SHAP at its core describes the average impact from adding a feature to a model, but does it in a way that attempts to account for all possible subsets of the other features as well. See [Lundberg et al.'s “From Local Explanations to Global Understanding with Explainable AI for Trees”](https://www.nature.com/articles/s42256-019-0138-9) for in-depth reading on SHAP values as applied to decision tree ensembles.\n",
+    "\n",
+    "In this notebook, we provide an example of training an XGBoost model with AWS SageMaker's [XGBoost estimator](https://docs.aws.amazon.com/sagemaker/latest/dg/xgboost.html), and then use SHAP values to identify key features and feature interactions in our dataset. SHAP values have been available in XGBoost for several versions already, but 1.3 brings GPU acceleration, reducing computation time by up to 20x for SHAP values and 340x for SHAP interaction values. This is powered under the hood by RAPIDS GPUTreeShap, which offers portable CUDA C++ implementations of SHAP algorithms for decision tree models. This algorithm is described in detail in [GPUTreeShap: Fast Parallel Tree Interpretability\n",
+    "](https://arxiv.org/abs/2010.13972). \n",
+    "\n",
+    "We will be using the California housing dataset. This is a famous dataset of house prices and attributes in California from the 1990 Census, available via [scikit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.datasets.fetch_california_housing.html).\n",
+    "\n",
+    "## 0. Preliminary Setup\n",
+    "\n",
+    "This notebook was tested in an Amazon [SageMaker Studio](https://docs.aws.amazon.com/sagemaker/latest/dg/studio.html) notebook, on a ml.t3.medium instance with Python 3 (Data Science) kernel. As a preliminary step, we first ensure that the latest version of SageMaker is installed: "
```

Comment:
Does `ml.t3.medium` have GPUs? You'd probably need to run this on a `ml.p3.2xlarge` 

Check here: https://docs.aws.amazon.com/sagemaker/latest/dg/notebooks-available-instance-types.html